bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java

```diff
@@ -601,4 +602,40 @@ public static String prepareQueryForLog(String query, int maxLength) {
         ? noNewLinesQuery.substring(0, maxLength) + /* ellipsis */ '\u2026'
         : noNewLinesQuery;
   }
+
+  static String getQueryForTimePartitionedTable(
+      String destinationTableName,
+      String temporaryTableName,
+      StandardTableDefinition destinationDefinition,
+      TimePartitioning timePartitioning) {
+    TimePartitioning.Type partitionType = timePartitioning.getType();
+    String partitionField = timePartitioning.getField();
+    String extractedPartitioned = "timestamp_trunc(`%s`.`%s`, %s)";
```

Comment:
What happens for fields of type `DATE` or `DATETIME`?

Reply:
timestamp_trunc works for `DATE` and `DATETIME` as well